modules/core/src/main/java/org/apache/ignite/internal/util/nio/ssl/GridNioSslHandler.java

```diff
@@ -227,10 +227,10 @@ void handshake() throws IgniteCheckedException, SSLException {
 
                             if (fut != null)
                                 fut.onDone();
-
-                            parent.proceedSessionOpened(ses);
                         }
 
+                        parent.proceedSessionOpened(ses);
```

Comment:
Why you change it?

Reply:
When ssl init handshake complete it doesn't proceed session and dont create compression handler. It processed through incoming messages where filters are reversed ( first SSL, second compress  etc)

Reply:
Add this information in comment, please.